runtime/gateway.go

```diff
@@ -89,25 +89,27 @@ type Options struct {
 
 // Gateway type
 type Gateway struct {
-	HTTPPort         int32
-	TChannelPort     int32
-	RealHTTPPort     int32
-	RealHTTPAddr     string
-	RealTChannelPort int32
-	RealTChannelAddr string
-	WaitGroup        *sync.WaitGroup
-	Channel          *tchannel.Channel
-	ContextLogger    ContextLogger
-	ContextMetrics   ContextMetrics
-	ContextExtractor ContextExtractor
-	RootScope        tally.Scope
-	Logger           *zap.Logger
-	ServiceName      string
-	Config           *StaticConfig
-	HTTPRouter       HTTPRouter
-	TChannelRouter   *TChannelRouter
-	Tracer           opentracing.Tracer
-	JSONWrapper      jsonwrapper.JSONWrapper
+	HTTPPort               int32
+	TChannelPort           int32
+	RealHTTPPort           int32
+	RealHTTPAddr           string
+	RealTChannelPort       int32
+	RealTChannelAddr       string
+	WaitGroup              *sync.WaitGroup
+	ServerTChannel         *tchannel.Channel
+	ClientTChannels        map[string]*tchannel.Channel
+	ContextLogger          ContextLogger
+	ContextMetrics         ContextMetrics
+	ContextExtractor       ContextExtractor
+	RootScope              tally.Scope
+	Logger                 *zap.Logger
+	ServiceName            string
+	Config                 *StaticConfig
+	HTTPRouter             HTTPRouter
+	ServerTChannelRouter   *TChannelRouter
+	TChannelSubLoggerLevel zapcore.Level
+	Tracer                 opentracing.Tracer
+	JSONWrapper            jsonwrapper.JSONWrapper
 
```

Comment:
The diff here might make it hard to understand what has changed in the struct. Refer below

```
Channel -> .ServerTChannel
TChannelRouter -> ServerTChannelRouter
+ TChannelSubLoggerLevel
+ ClientTChannels
```